core/src/main/kotlin/com/alecstrong/sql/psi/core/psi/SchemaContributor.kt

```diff
@@ -0,0 +1,12 @@
+package com.alecstrong.sql.psi.core.psi
+
+import com.intellij.util.containers.MultiMap
+import kotlin.reflect.KClass
+
+internal interface SchemaContributor {
+  fun modifySchema(schema: Schema): Schema
+}
+
+internal data class Schema(
+  val types: Map<KClass<*>, MultiMap<String, Any>>
```

Comment:
Could we use more specific types here? Something like [this](https://github.com/AlecStrong/sql-psi/compare/astrong/2020-09-17/schema-contributor...veyndan:astrong/2020-09-17/schema-contributor?expand=1):

```kotlin
val types: Map<KClass<out SchemaContributor>, MultiMap<String, SchemaContributor>>
```

Reply:
I might try to use this for tables as well, in that case its `LazyQuery` which doesn't extend `SchemaContributor`

that said i think making `types` value smarter so that its more like `Map<T, MultiMap<String, T>>` is a good idea, will fix that

Reply:
Cool, makes sense